c++/src/BlockBuffer.cc

```diff
@@ -82,4 +83,42 @@ namespace orc {
       }
     }
   }
+
+  uint64_t BlockBuffer::writeTo(OutputStream* output) {
+    static uint64_t MAX_CHUNK_SIZE = 1024 * 1024 * 1024;
+    uint64_t chunkSize = std::min(output->getNaturalWriteSize(), MAX_CHUNK_SIZE);
+    if (chunkSize == 0) {
+      throw std::logic_error("Natural write size cannot be zero");
+    }
+    char* chunk = memoryPool.malloc(chunkSize);
```

Comment:
If blockNumber == 1, we should avoid this unnecessary allocation

Reply:
if natural write size less than block size, here need to malloc chunk.

Reply:
Yes, however you can still get rid of the allocation when blockNumber == 1 && naturalWriteSize >= block.size. BTW, we can keep the current implementation for now

Reply:
Allocation can be avoided here when blockNumber == 1 && currentSize <= NaturalWriteSize.